File: app/src/main/java/com/caramelheaven/lennach/models/mapper/Mapper.java
package com.caramelheaven.lennach.models.mapper;

/**
 * Created by CaramelHeaven on 20:54, 23/01/2019.
 */
public abstract class Mapper<T, T1> {
    protected abstract T map(T1 value);

    protected abstract void fillData(T myData, T1 values);
}

File: app/src/main/java/com/caramelheaven/lennach/models/mapper/Mapper.kt
package com.caramelheaven.lennach.models.mapper

/**
 * Created by CaramelHeaven on 16:26, 03/02/2019.
 *
 * @param T  - output class
 * @param T1 - enter class
 */
abstract class Mapper<T, T1> {
    abstract fun map(enterData: T1): T

    protected abstract fun fillData(outputData: T, enterData: T1)
}
